// @angular/forms/src/directives/ng_model_group.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/forms/src/directives/ng_model_group');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import3 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgModelGroup = (function () {
    function Wrapper_NgModelGroup(p0, p1, p2) {
        this._changed = false;
        this.context = new import0.NgModelGroup(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgModelGroup.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_NgModelGroup.prototype.ngOnDestroy = function () {
    };
    Wrapper_NgModelGroup.prototype.check_name = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.name = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgModelGroup.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_NgModelGroup.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_NgModelGroup.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_NgModelGroup.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_NgModelGroup;
}());
exports.Wrapper_NgModelGroup = Wrapper_NgModelGroup;
