// @angular/forms/src/directives/range_value_accessor.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/forms/src/directives/range_value_accessor');
var Wrapper_RangeValueAccessor = (function () {
    function Wrapper_RangeValueAccessor(p0, p1) {
        this._changed = false;
        this.context = new import0.RangeValueAccessor(p0, p1);
    }
    Wrapper_RangeValueAccessor.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_RangeValueAccessor.prototype.ngOnDestroy = function () {
    };
    Wrapper_RangeValueAccessor.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_RangeValueAccessor.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_RangeValueAccessor.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'change')) {
            var pd_sub_0 = (this.context.onChange($event.target.value) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'input')) {
            var pd_sub_1 = (this.context.onChange($event.target.value) !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'blur')) {
            var pd_sub_2 = (this.context.onTouched() !== false);
            result = (pd_sub_2 && result);
        }
        return result;
    };
    Wrapper_RangeValueAccessor.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_RangeValueAccessor;
}());
exports.Wrapper_RangeValueAccessor = Wrapper_RangeValueAccessor;
